inference-engine/src/readers/ir_reader_v7/ie_ir_parser.cpp

```diff
@@ -53,21 +53,14 @@ std::shared_ptr<ICNNNetwork> CNNParser::parse(const pugi::xml_node& root, std::i
     StatusCode ret = reader.ReadNetwork(root, &resp);
     if (ret != OK)
         THROW_IE_EXCEPTION << resp.msg;
+
     TBlob<uint8_t>::Ptr weightsPtr;
 
-    // Try to get BlobStream to work with original blob
-    details::BlobStream* blobStream = getBlobStream(binStream);
-    if (blobStream != nullptr) {
-        weightsPtr = std::make_shared<WeightsHolderBlob>(blobStream->getBlob());
+    if (weights != nullptr) {
+        weightsPtr = TBlob<uint8_t>::Ptr(new WeightsHolderBlob(weights));
     } else {
-        // Allocate a blob for weights
-        binStream.seekg(0, std::ios::end);
-        size_t length = binStream.tellg();
-        weightsPtr = std::make_shared<TBlob<uint8_t>>(TensorDesc(Precision::U8, {length}, Layout::C));
+        weightsPtr = std::make_shared<TBlob<uint8_t>>(TensorDesc(Precision::U8, { 0 }, Layout::C));
```

Comment:
I think you should pass empty `weightsPtr` without this fake creation.

Reply:
in another case test without weights crashes with exception while  trying to access blob. In previous implementation it passes empty stream with 0 length, so i did  the same - zero size blob.

Reply:
Ok got it.